Nitrox.Model/Extensions/TaskExtensions.cs

```diff
@@ -5,6 +5,8 @@ namespace Nitrox.Model.Extensions;
 
 public static class TaskExtensions
 {
+    public static bool IsBusyOrDone(this Task? task) => task is { IsCompleted: true, IsFaulted: false, IsCanceled: false };
```

Comment:
Maybe I am misunderstanding this, but imo this returns true if the task is busy (working) or done. Wouldn't IsCompleted == true not reflect the busy part?